packages/compass-indexes/src/components/regular-indexes-table/regular-indexes-table.tsx

```diff
@@ -163,44 +307,40 @@ export const RegularIndexesTable: React.FunctionComponent<
     };
   }, [tabId, onRegularIndexesOpened, onRegularIndexesClosed]);
 
+  const allIndexes: MergedIndex[] = mergeIndexes(indexes, inProgressIndexes);
+
   const data = useMemo<LGTableDataType<IndexInfo>[]>(
     () =>
-      indexes.map((index) => ({
-        id: index.name,
-        name: index.name,
-        indexInfo: index,
-        type: <TypeField type={index.type} extra={index.extra} />,
-        size: <SizeField size={index.size} relativeSize={index.relativeSize} />,
-        usage: <UsageField usage={index.usageCount} since={index.usageSince} />,
-        properties: (
-          <PropertyField
-            cardinality={index.cardinality}
-            extra={index.extra}
-            properties={index.properties}
-          />
-        ),
-        actions: index.name !== '_id_' &&
-          index.extra.status !== 'inprogress' && (
-            <IndexActions
-              index={index}
-              serverVersion={serverVersion}
-              onDeleteIndexClick={onDeleteIndexClick}
-              onHideIndexClick={onHideIndexClick}
-              onUnhideIndexClick={onUnhideIndexClick}
-            ></IndexActions>
-          ),
+      allIndexes.map((index) => {
+        let indexData: CommonIndexInfo;
+        if (index.compassIndexType === 'in-progress-index') {
+          indexData = getInProgressIndexInfo(index, {
+            onDeleteFailedIndexClick,
+          });
+        } else {
+          indexData = getRegularIndexInfo(index, {
+            serverVersion,
+            onDeleteIndexClick,
+            onHideIndexClick,
+            onUnhideIndexClick,
+          });
+        }
 
-        // eslint-disable-next-line react/display-name
-        renderExpandedContent: () => (
-          <IndexKeysBadge
-            keys={index.fields}
-            data-testid={`indexes-details-${index.name}`}
-          />
-        ),
-      })),
+        return {
+          ...indexData,
+          // eslint-disable-next-line react/display-name
+          renderExpandedContent: () => (
```

Comment:
Super tiny nit: if you write this as a named function, you won't need to eslint-disable:

```jsx
          renderExpandedContent() { return (
            // ...
```

Reply:
Oops. Never even noticed that. I just copy/pasted what we had :) Will fix.